test/runtests.jl

```diff
@@ -134,49 +151,216 @@ end
-        (10, "IRAN", 4, "furiously idle platelets nag. express asymptotes s"),
-        (11, "IRAQ", 4, "pendencies; slyly express foxes integrate carefully across the reg"),
-        (12, "JAPAN", 2, " quickly final packages. furiously i"),
-        (13, "JORDAN", 4, "the slyly regular ideas. silent Tiresias affix slyly fu"),
-        (14, "KENYA", 0, "lyly special foxes. slyly regular deposits sleep carefully. carefully permanent accounts slee"),
-        (15, "MOROCCO", 0, "ct blithely: blithely express accounts nag carefully. silent packages haggle carefully abo"),
-        (16, "MOZAMBIQUE", 0, " beans after the carefully regular accounts r"),
-        (17, "PERU", 1, "ly final foxes. blithely ironic accounts haggle. regular foxes about the regular deposits are furiously ir"),
-        (18, "CHINA", 2, "ckly special packages cajole slyly. unusual, unusual theodolites mold furiously. slyly sile"),
-        (19, "ROMANIA", 3, "sly blithe requests. thinly bold deposits above the blithely regular accounts nag special, final requests. care"),
-        (20, "SAUDI ARABIA", 4, "se slyly across the blithely regular deposits. deposits use carefully regular "),
-        (21, "VIETNAM", 2, "lly across the quickly even pinto beans. caref"),
-        (22, "RUSSIA", 3, "uctions. furiously unusual instructions sleep furiously ironic packages. slyly "),
-        (23, "UNITED KINGDOM", 3, "carefully pending courts sleep above the ironic, regular theo"),
-        (24, "UNITED STATES", 1, "ly ironic requests along the slyly bold ideas hang after the blithely special notornis; blithely even accounts")
-    ]
+@testset "Incremental Scan API" begin
+    println("Testing incremental scan API...")
 
-    println("✅ Nations table read and verified successfully")
+    # Use the test table created specifically for incremental scan testing
+    test_snapshot_path = "s3://warehouse/incremental/test1/metadata/00003-359e8bb8-1e5d-46d2-bcde-fdaeaa41114f.metadata.json"
+
+    # Open the table
+    table = RustyIceberg.table_open(test_snapshot_path)
+    @test table != C_NULL
+    println("✅ Table opened successfully")
+
+    # Use real snapshot IDs from the test table
+    from_snapshot_id = Int64(6540713100348352610)
+    to_snapshot_id = Int64(6832180054960511692)
+
+    # The table is created with these transactions. Above snapshot IDs are for the 2nd and the 4th txn below:
+    #=
+        CREATE TABLE demo.incremental.test1 USING iceberg
+        TBLPROPERTIES ('write.delete.mode' = 'merge-on-read')
+        AS (SELECT n FROM range(1, 11) r(n));
+
+        INSERT INTO incremental.test1
+        SELECT n FROM range(101, 200) r(n);
+
+        INSERT INTO incremental.test1
+        SELECT n FROM range(201, 300) r(n);
+
+        DELETE FROM incremental.test1 WHERE n = 150 OR n = 250;
+    =#
+
+    @testset "Incremental Scan E2E Test" begin
```

Comment:
Should we add a test that exercises some of the `with_...` functions and column selection?